src/pybind/mgr/cephadm/schedule.py

```diff
@@ -124,6 +124,23 @@ def place(self):
             logger.debug('Provided hosts: %s' % candidates)
             return candidates
 
+        # if asked to place even number of mons, deploy 1 less
```

Comment:
I think we not _only_ need to talk about `count`.

The initial tracker defines it a bit broader with:

```
In case you have 4 hosts, cephadm will by default spawn 4 MONs.

We should probably improve this and only create 3 MONs.

```

That'd mean we need to correct the number of mons for:

* hosts
* host_placements
* count
* labels


With the current implementation this:

``` yaml
service_type: mon
placement:
  hosts:
    - host1
    - host2
    - host3
    - host4
```

would still deploy four nodes

Reply:
@jschmid1 this PR should now work with placements specs using other methods than count.